server/modules/surveyRdb/manager/surveyRdbManager.js

```diff
@@ -100,22 +102,35 @@ export const queryTable = async (
       rows.map(async (row) => {
         const recordUuid = row[DataTable.colNameRecordUuuid]
         const record = await RecordRepository.fetchRecordByUuid(surveyId, recordUuid)
-        const parentNodeUuid = R.prop(`${NodeDef.getName(nodeDefTable)}_uuid`, row)
+        const parentNodeUuid = R.prop(RdbDataView.getColUuid(nodeDefTable), row)
         const resultRow = { ...row, cols: {}, record, parentNodeUuid }
 
         // Assoc nodes to each columns
         await PromiseUtils.each(nodeDefUuidCols, async (nodeDefUuidCol) => {
           const nodeDefCol = Survey.getNodeDefByUuid(nodeDefUuidCol)(survey)
           const nodeDefColParent = Survey.getNodeDefParent(nodeDefCol)(survey)
-          const parentUuidColName = `${NodeDef.getName(nodeDefColParent)}_uuid`
+          const parentUuidColName = RdbDataView.getColUuid(nodeDefColParent)
           const parentUuid = R.prop(parentUuidColName, row)
 
-          const node =
-            NodeDef.isMultiple(nodeDefTable) && NodeDef.isEqual(nodeDefCol)(nodeDefTable) // Column is the multiple attribute
-              ? await NodeRepository.fetchNodeByUuid(surveyId, row[`${NodeDef.getName(nodeDefCol)}_uuid`])
-              : (
-                  await NodeRepository.fetchChildNodesByNodeDefUuids(surveyId, recordUuid, parentUuid, [nodeDefUuidCol])
-                )[0]
+          let node = null
+          if (NodeDef.isAnalysis(nodeDefCol)) {
+            // Fetch node from node results table
+            node = R.head(
+              await ResultNodeRepository.fetchNodeResultsByRecordAndNodeDefUuid({
+                surveyId,
+                recordUuid,
+                parentUuid,
+                nodeDefUuid: nodeDefUuidCol,
+              })
+            )
+          } else if (NodeDef.isMultiple(nodeDefTable) && NodeDef.isEqual(nodeDefCol)(nodeDefTable)) {
+            // Column is the multiple attribute
+            node = await NodeRepository.fetchNodeByUuid(surveyId, row[RdbDataView.getColUuid(nodeDefCol)])
+          } else {
+            node = R.head(
+              await NodeRepository.fetchChildNodesByNodeDefUuids(surveyId, recordUuid, parentUuid, [nodeDefUuidCol])
+            )
+          }
```

Comment:
can we do just two fetches here? one from node table and one from resultNode table?

Reply:
they are a bit different, one fetch is by node uuid, the other is by parent node uuid and node def uuids...